examples/platform/silabs/efr32/BUILD.gn

```diff
@@ -259,9 +261,9 @@ source_set("efr32-common") {
     "${efr32_sdk_build_root}:silabs_config",
   ]
 
-  # MQTT
+  # DIC
   if (enable_dic) {
-    deps += [ "${examples_plat_dir}:efr32-dic" ]
+    deps += [ "${examples_plat_dir}:silabs-dic" ]
```

Comment:
Changing this to `public_deps` instead should make it so you don't have to add the dep in each example `build.gn `(like you do in this pr for `lighting-app/silabs/build.gn` and  `lock-app/silabs/build.gn`)

Reply:
@jmartinez-silabs  Modified changes as per you suggestions for Build.gn of light and lock app by doing public_deps for dic.